pkg/operator/sync_openshiftcontrollermanager_v311_00.go

```diff
@@ -43,63 +43,100 @@ func syncOpenShiftControllerManager_v311_00_to_latest(
+		if clientCAConfigMap != nil { // SyncConfigMap can return nil
+			resourceVersion = clientCAConfigMap.ObjectMeta.ResourceVersion
+		}
+		specAnnotations["configmaps/client-ca"] = resourceVersion
 	}
 
-	_, serviceCAModified, err := manageOpenShiftServiceCAConfigMap_v311_00_to_latest(c.kubeClient, c.configMapsGetter, c.recorder)
+	serviceCAConfigMap, _, err := manageOpenShiftServiceCAConfigMap_v311_00_to_latest(c.kubeClient, c.configMapsGetter, c.recorder)
 	if err != nil {
 		errors = append(errors, fmt.Errorf("%q %q: %v", operandName, "openshift-service-ca", err))
+	} else {
+		specAnnotations["configmaps/openshift-service-ca"] = serviceCAConfigMap.ObjectMeta.ResourceVersion
 	}
 
-	_, globalCAModified, err := manageOpenShiftGlobalCAConfigMap_v311_00_to_latest(c.kubeClient, c.configMapsGetter, c.recorder)
+	globalCAConfigMap, _, err := manageOpenShiftGlobalCAConfigMap_v311_00_to_latest(c.kubeClient, c.configMapsGetter, c.recorder)
 	if err != nil {
 		errors = append(errors, fmt.Errorf("%q %q: %v", operandName, "openshift-global-ca", err))
+	} else {
+		specAnnotations["configmaps/openshift-global-ca"] = globalCAConfigMap.ObjectMeta.ResourceVersion
 	}
 
 	// Route Controller Manager
-	_, rcConfigMapModified, err := manageRouteControllerManagerConfigMap_v311_00_to_latest(c.kubeClient, c.configMapsGetter, c.recorder, operatorConfig)
+	rcConfigMap, _, err := manageRouteControllerManagerConfigMap_v311_00_to_latest(c.kubeClient, c.configMapsGetter, c.recorder, operatorConfig)
 	if err != nil {
 		errors = append(errors, fmt.Errorf("%q %q: %v", rcOperandName, "configmap", err))
+	} else {
+		rcSpecAnnotations["configmaps/config"] = rcConfigMap.ObjectMeta.ResourceVersion
 	}
 
 	// the kube-apiserver is the source of truth for client CA bundles
-	rcClientCAModified, err := manageRouteControllerManagerClientCA_v311_00_to_latest(c.kubeClient.CoreV1(), c.recorder)
+	rcClientCAConfigMap, _, err := manageRouteControllerManagerClientCA_v311_00_to_latest(c.kubeClient.CoreV1(), c.recorder)
 	if err != nil {
 		errors = append(errors, fmt.Errorf("%q %q: %v", rcOperandName, "client-ca", err))
+	} else {
+		resourceVersion := "0"
+		if rcClientCAConfigMap != nil { // SyncConfigMap can return nil
+			resourceVersion = rcClientCAConfigMap.ObjectMeta.ResourceVersion
+		}
+		rcSpecAnnotations["configmaps/client-ca"] = resourceVersion
 	}
 
-	forceRollout = forceRollout || operatorConfig.ObjectMeta.Generation != operatorConfig.Status.ObservedGeneration
-	forceRollout = forceRollout || configMapModified || clientCAModified || serviceCAModified || globalCAModified
-
-	rcForceRollout = rcForceRollout || operatorConfig.ObjectMeta.Generation != operatorConfig.Status.ObservedGeneration
-	rcForceRollout = rcForceRollout || rcConfigMapModified || rcClientCAModified
-
-	// our configmaps and secrets are in order, now it is time to create the DS
+	// our configmaps and secrets are in order, now it is time to create the Deployment
 	// TODO check basic preconditions here
```

Comment:
@deads2k Do you remember the story behind this this TODO and if it is still relevant?

Reply:
I don't remember

Reply:
removed the TODO then, I hope that is okay